Problems/1065/ans_1065_y1.py

```diff
@@ -1,29 +1,26 @@
 #https://www.acmicpc.net/problem/1065
 
 #각 자리수의 등차 수열 : 한자리 수 - 1개 / 두자리 수 - 1개 / 세자리 수 - 계산 필요
-#abc의 등차 수열 전제 조건: a + 2d (=c) < 10 , a + d(=b) < 10
-#전제 조건 재정리 : 111a+12d <= abc || 89a-12d <=abc , d <= 4
+#abc의 등차 수열1(d:양수): a + 2d (=c) < 10 -> 100의 자리 바뀔 때마다 +4
+#abc의 등차 수열2(d:음수): a - 2d >=0 -> a/2
 
 
-def equal_diff(N):
-    if N < 100:
-        return N
+def AP(MAX):
+    cnt=0
+    n = str(MAX)
+    
+    if (len(n)<3):
+        cnt = MAX
     else:
         cnt = 99
-        N_1st = (N//10)//10
-        
-        for i in range(1,N_1st+1):
-            for j in range(1,5):
-                if (111*i+12*j) <= N:
-                    cnt+=1
-                    print(cnt)
-                if (111*i+12*j) <= N:
-                    cnt+=1
-                    print(cnt)
-        return cnt               
+        for i in range(100,MAX+1):
+            str_i = str(i)
+            if ((int(str_i[0])-int(str_i[1]))==(int(str_i[1])-int(str_i[2]))):
```

Comment:
```suggestion
            str_i = list(map(int,list(str(i))))
            if str_i[0]-str_i[1]==str_i[1]-str_i[2]:
```
자주 쓰는 string 숫자 integer list로 분해 방법

Reply:
홀... 신박... 감사..ㅋㅋ